File: backend/src/main/java/com/apocalypse/thefall/repository/character/stats/SpecialRepository.java
package com.apocalypse.thefall.repository.character.stats;

import com.apocalypse.thefall.entity.character.stats.Special;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface SpecialRepository extends JpaRepository<Special, Long> {

    List<Special> findAllByVisibleTrueOrderByDisplayOrderAsc();
}

File: backend/src/main/java/com/apocalypse/thefall/repository/character/stats/SpecialRepository.kt
package com.apocalypse.thefall.repository.character.stats

import com.apocalypse.thefall.entity.character.stats.Special
import jakarta.persistence.QueryHint
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.QueryHints

interface SpecialRepository : JpaRepository<Special, Long> {
    @QueryHints(QueryHint(name = "org.hibernate.cacheable", value = "true"))
    fun findAllByVisibleTrueOrderByDisplayOrderAsc(): List<Special>
}
